lib/src/components/NxTabs/stateful/NxStatefulTabs.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { cloneElement, isValidElement, useState, Children } from 'react';
+
+import NxTabs from '../NxTabs';
+import { Props as NxTabProps } from '../../NxTab/types';
+
+import { Props, propTypes } from './types';
+
+/**
+ * A stateful component for rendering clickable tabs.
+ * Clicking on a NxTabLabel will show the associated Tab (determined by using an index number for each).
+ * This assumes the following:
+ * * The first child must be a NxTabList component.
+ * * There is at least one NxTabPanel component for each NxTab in the NxTabList component.
+ * * The NxTabPanel must have a labelledBy prop that matches the id of an NxTab.
+ */
+const NxStatefulTabs = function NxStatefulTabsElement(props: Props) {
+  const { children, ...attrs } = props;
+  const [originalTabList, ...tabPanels] = Children.toArray(children);
+
+  if (!isValidElement(originalTabList)) {
+    return null;
```

Comment:
Why is this check here and not in `NxTabs`?

Reply:
> Why is this check here and not in NxTabs?

Further is silent failure the best approach here?

Reply:
I added this check because the `NxStatefulTabs` needs to be sure that there is at least one valid tab to prevent the code from breaking. It's extra logic that didn't seem necessary in `NxTabs` as the code wouldn't _break_ if you didn't include the correct children, it just wouldn't do anything in particular. If we wanted to I suppose we could move this check down to `NxTabs` to guide users to the correct usage of the component, but there would be a small performance cost to doing that.

Reply:
6fe2aeb